### PR TITLE
feat: Shared Data files (Feature PR)

### LINKED ISSUE #313: Shared data files — display
## Description
Component to display shared files within a group.

### LINKED ISSUE #314: Shared data files — upload
## Description
Component to upload new files.

* When multiple files are supplies, upload one at a time.
* If one of the files is rejected, other files are still accepted.

### LINKED ISSUE #315: Shared data files — modification
## Description
Component to rename and delete files and change their descriptions.\

* When file owner deletes a file, it is removed
* When group manager deletes a file, detach relationship between file and group

### PR DESCRIPTION
**Pull request by josebui**
## Description

- Complete Shared data files feature

### Checklist
- [x] Corresponding issue has been opened
- [x] New tests added

### Related Issues
Fixes #313, Fixes #314, Fixes #315.

### CODE REVIEW

File: src/localization/locales/en-US.json [hunk 1]
```diff
@@ -244,7 +244,34 @@
         "form_boundary_geojson_title": "Upload your landscape’s boundary",
         "form_location_placeholder": "Landscape location",
         "form_location_select": "Select location",
-        "form_name_placeholder": "Your landscape’s official name"
+        "form_name_placeholder": "Your landscape’s official name",
+        "boundaries_file_no_accepted": "[TODO] No accepted files",
```
Comment (Hanal-code) [inline src/localization/locales/en-US.json:248]: The file(s) cannot be added.

File: src/localization/locales/en-US.json [hunk 2]
```diff
@@ -244,7 +244,34 @@
...
+    "shared_data": {
+        "title": "Shared files",
+        "description": "Only the members of your group can download the files. <1>Read more about sharing files</1>.",
+        "learn_more_url": "http://terraso.org/help/sharing-files/",
+        "upload_button": "Share Files",
+        "delete_confirm_title": "Delete “{{name}}”?",
+        "delete_confirm_message": "Are you sure you wish to delete the file “{{name}}”?",
+        "delete_confirm_button": "Delete File",
+        "deleted": "The file “{{name}}” was deleted.",
+        "upload_file_success": "[TODO] File upload succesfuly",
```
Comment (Hanal-code) [inline src/localization/locales/en-US.json:260]: Uploaded successfully.

File: src/localization/locales/en-US.json [hunk 3]
```diff
@@ -244,7 +244,34 @@
...
+        "upload_filename_label": "File Name",
+        "upload_description_label": "Description",
+        "upload_description_placeholder": "[TODO] e.g. Soil health survey data",
```
Comment (Hanal-code) [inline src/localization/locales/en-US.json:263]: e.g. Soil health survey data

File: src/localization/locales/en-US.json [hunk 4]
```diff
@@ -244,7 +244,34 @@
...
+        "upload_no_files": "No files added",
+        "upload_rejected_too-many-files": "[TODO] Too many files. Max number of files: {{maxFiles}}",
+        "upload_rejected_file-invalid-type": "[TODO] Invalid file type. Accepted extensions: {{fileExtensions}}. Files: {{rejectedFiles}}",
```
Comment (Hanal-code) [inline src/localization/locales/en-US.json:266]: {{rejectedFiles}} cannot be added because the file type(s) are not supported.

**Review by cacarrara — APPROVED** [review]

**Pull request merged**